src/zabapgit_object_sxci.prog.abap

```diff
@@ -0,0 +1,328 @@
+          lt_filters              TYPE seex_filter_table,
+          lt_methods              TYPE seex_mtd_table.
+
+    lv_implementation_name = ms_item-obj_name.
+
+    CALL FUNCTION 'SXV_EXIT_FOR_IMP'
+      EXPORTING
+        imp_name           = lv_implementation_name
+      IMPORTING
+        exit_name          = lv_exit_name
+      TABLES
+        filters            = lt_filters
+      EXCEPTIONS
+        data_inconsistency = 1
+        OTHERS             = 2.
+
+    IF sy-subrc <> 0.
+      zcx_abapgit_exception=>raise( 'error from SXV_EXIT_FOR_IMP' ).
+    ENDIF.
+
+    CALL FUNCTION 'SXO_BADI_READ'
+      EXPORTING
+        exit_name    = lv_exit_name
+      IMPORTING
+        badi         = ls_badi_definition
+        filter_obj   = lo_filter_object
+      TABLES
+        fcodes       = lt_function_codes
+        cocos        = lt_control_composites
+        intas        = lt_customer_includes
+        scrns        = lt_screens
+        methods      = lt_methods
+      EXCEPTIONS
+        read_failure = 1
+        OTHERS       = 2.
+
+    IF sy-subrc <> 0.
+      zcx_abapgit_exception=>raise( 'error from SXV_EXIT_FOR_IMP' ).
+    ENDIF.
+
+    CALL FUNCTION 'SXO_IMPL_FOR_BADI_READ'
+      EXPORTING
+        imp_name          = lv_implementation_name
+        exit_name         = lv_exit_name
+        inter_name        = ls_badi_definition-inter_name
+        filter_obj        = lo_filter_object
+      IMPORTING
+        impl              = ls_badi_implementation
+        filter_values_obj = lo_filter_values_object
+      TABLES
+        fcodes            = lt_function_codes
+        cocos             = lt_control_composites
+        intas             = lt_customer_includes
+        scrns             = lt_screens
+      CHANGING
+        methods           = lt_methods
+      EXCEPTIONS
+        read_failure      = 1
+        OTHERS            = 2.
+
+    IF sy-subrc <> 0.
+      zcx_abapgit_exception=>raise( 'error from SXV_EXIT_FOR_IMP' ).
+    ENDIF.
+
+    CLEAR: ls_badi_implementation-aname,
+           ls_badi_implementation-adate,
+           ls_badi_implementation-atime,
+           ls_badi_implementation-uname,
+           ls_badi_implementation-udate,
+           ls_badi_implementation-utime,
+           ls_badi_implementation-active.
+
+    io_xml->add( iv_name = 'SXCI'
+                 ig_data = ls_badi_implementation ).
+
+    io_xml->add( iv_name = co_badi_component_name-filters
+                 ig_data = lt_filters ).
+
+    io_xml->add( iv_name = co_badi_component_name-function_codes
+                 ig_data = lt_function_codes ).
+
+    io_xml->add( iv_name = co_badi_component_name-control_composites
+                 ig_data = lt_control_composites ).
+
+    io_xml->add( iv_name = co_badi_component_name-customer_includes
+                 ig_data = lt_customer_includes ).
+
+    io_xml->add( iv_name = co_badi_component_name-screens
+                 ig_data = lt_screens ).
```

Comment:
can we put lt_filters + lt_function_codes + lt_control_composites + lt_customer_includes + lt_screens into a structure? this way it is easier to extend, plus less code as io_xml->add will only be called once